<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function utilizes the Seurat package to perform a differential expression (DE) test on gene expression data.
Users have the flexibility to specify custom cell groups, marker types, and various options for DE analysis."><title>Differential gene test — RunDEtest • SCP</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential gene test — RunDEtest"><meta property="og:description" content="This function utilizes the Seurat package to perform a differential expression (DE) test on gene expression data.
Users have the flexibility to specify custom cell groups, marker types, and various options for DE analysis."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SCP</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.5.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/zhanghao-njmu/SCP">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Differential gene test</h1>
      <small class="dont-index">Source: <a href="https://github.com/zhanghao-njmu/SCP/blob/HEAD/R/SCP-analysis.R" class="external-link"><code>R/SCP-analysis.R</code></a></small>
      <div class="d-none name"><code>RunDEtest.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function utilizes the Seurat package to perform a differential expression (DE) test on gene expression data.
Users have the flexibility to specify custom cell groups, marker types, and various options for DE analysis.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">RunDEtest</span><span class="op">(</span></span>
<span>  <span class="va">srt</span>,</span>
<span>  group_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group1 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cells1 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  cells2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  features <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  markers_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"all"</span>, <span class="st">"paired"</span>, <span class="st">"conserved"</span>, <span class="st">"disturbed"</span><span class="op">)</span>,</span>
<span>  grouping.var <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  meta.method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"maximump"</span>, <span class="st">"minimump"</span>, <span class="st">"wilkinsonp"</span>, <span class="st">"meanp"</span>, <span class="st">"sump"</span>, <span class="st">"votep"</span><span class="op">)</span>,</span>
<span>  test.use <span class="op">=</span> <span class="st">"wilcox"</span>,</span>
<span>  only.pos <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  fc.threshold <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  base <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  pseudocount.use <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  mean.fxn <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min.pct <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  min.diff.pct <span class="op">=</span> <span class="op">-</span><span class="cn">Inf</span>,</span>
<span>  max.cells.per.ident <span class="op">=</span> <span class="cn">Inf</span>,</span>
<span>  latent.vars <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  min.cells.feature <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  min.cells.group <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  norm.method <span class="op">=</span> <span class="st">"LogNormalize"</span>,</span>
<span>  p.adjust.method <span class="op">=</span> <span class="st">"bonferroni"</span>,</span>
<span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span>
<span>  assay <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">bpparam</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">11</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>srt</dt>
<dd><p>A Seurat object.</p></dd>


<dt>group_by</dt>
<dd><p>A grouping variable in the dataset to define the groups or conditions for the differential test. If not provided, the function uses the "active.ident" variable in the Seurat object.</p></dd>


<dt>group1</dt>
<dd><p>A vector of cell IDs or a character vector specifying the cells that belong to the first group. If both group_by and group1 are provided, group1 takes precedence.</p></dd>


<dt>group2</dt>
<dd><p>A vector of cell IDs or a character vector specifying the cells that belong to the second group. This parameter is only used when group_by or group1 is provided.</p></dd>


<dt>cells1</dt>
<dd><p>A vector of cell IDs specifying the cells that belong to group1. If provided, group1 is ignored.</p></dd>


<dt>cells2</dt>
<dd><p>A vector of cell IDs specifying the cells that belong to group2. This parameter is only used when cells1 is provided.</p></dd>


<dt>features</dt>
<dd><p>A vector of gene names specifying the features to consider for the differential test. If not provided, all features in the dataset are considered.</p></dd>


<dt>markers_type</dt>
<dd><p>A character value specifying the type of markers to find. Possible values are "all", "paired", "conserved", and "disturbed".</p></dd>


<dt>grouping.var</dt>
<dd><p>A character value specifying the grouping variable for finding conserved or disturbed markers. This parameter is only used when markers_type is "conserved" or "disturbed".</p></dd>


<dt>meta.method</dt>
<dd><p>A character value specifying the method to use for combining p-values in the conserved markers test. Possible values are "maximump", "minimump", "wilkinsonp", "meanp", "sump", and "votep".</p></dd>


<dt>test.use</dt>
<dd><p>Denotes which test to use. Available options are:</p><ul><li><p>"wilcox" : Identifies differentially expressed genes between two
 groups of cells using a Wilcoxon Rank Sum test (default)</p></li>
<li><p>"bimod" : Likelihood-ratio test for single cell gene expression,
 (McDavid et al., Bioinformatics, 2013)</p></li>
<li><p>"roc" : Identifies 'markers' of gene expression using ROC analysis.
 For each gene, evaluates (using AUC) a classifier built on that gene alone,
 to classify between two groups of cells. An AUC value of 1 means that
 expression values for this gene alone can perfectly classify the two
 groupings (i.e. Each of the cells in cells.1 exhibit a higher level than
 each of the cells in cells.2). An AUC value of 0 also means there is perfect
 classification, but in the other direction. A value of 0.5 implies that
 the gene has no predictive power to classify the two groups. Returns a
 'predictive power' (abs(AUC-0.5) * 2) ranked matrix of putative differentially
 expressed genes.</p></li>
<li><p>"t" : Identify differentially expressed genes between two groups of
 cells using the Student's t-test.</p></li>
<li><p>"negbinom" : Identifies differentially expressed genes between two
  groups of cells using a negative binomial generalized linear model.
  Use only for UMI-based datasets</p></li>
<li><p>"poisson" : Identifies differentially expressed genes between two
  groups of cells using a poisson generalized linear model.
  Use only for UMI-based datasets</p></li>
<li><p>"LR" : Uses a logistic regression framework to determine differentially
 expressed genes. Constructs a logistic regression model predicting group
 membership based on each feature individually and compares this to a null
 model with a likelihood ratio test.</p></li>
<li><p>"MAST" : Identifies differentially expressed genes between two groups
 of cells using a hurdle model tailored to scRNA-seq data. Utilizes the MAST
 package to run the DE testing.</p></li>
<li><p>"DESeq2" : Identifies differentially expressed genes between two groups
 of cells based on a model using DESeq2 which uses a negative binomial
 distribution (Love et al, Genome Biology, 2014).This test does not support
 pre-filtering of genes based on average difference (or percent detection rate)
 between cell groups. However, genes may be pre-filtered based on their
 minimum detection rate (min.pct) across both cell groups. To use this method,
 please install DESeq2, using the instructions at
 https://bioconductor.org/packages/release/bioc/html/DESeq2.html</p></li>
</ul></dd>


<dt>only.pos</dt>
<dd><p>Only return positive markers (FALSE by default)</p></dd>


<dt>fc.threshold</dt>
<dd><p>A numeric value used to filter genes for testing based on their average fold change between/among the two groups. By default, it is set to 1.5</p></dd>


<dt>base</dt>
<dd><p>The base with respect to which logarithms are computed.</p></dd>


<dt>pseudocount.use</dt>
<dd><p>Pseudocount to add to averaged expression values when
calculating logFC. 1 by default.</p></dd>


<dt>mean.fxn</dt>
<dd><p>Function to use for fold change or average difference calculation.
If NULL, the appropriate function will be chose according to the slot used</p></dd>


<dt>min.pct</dt>
<dd><p>only test genes that are detected in a minimum fraction of
min.pct cells in either of the two populations. Meant to speed up the function
by not testing genes that are very infrequently expressed. Default is 0.1</p></dd>


<dt>min.diff.pct</dt>
<dd><p>only test genes that show a minimum difference in the
fraction of detection between the two groups. Set to -Inf by default</p></dd>


<dt>max.cells.per.ident</dt>
<dd><p>Down sample each identity class to a max number.
Default is no downsampling. Not activated by default (set to Inf)</p></dd>


<dt>latent.vars</dt>
<dd><p>Variables to test, used only when <code>test.use</code> is one of
'LR', 'negbinom', 'poisson', or 'MAST'</p></dd>


<dt>min.cells.feature</dt>
<dd><p>Minimum number of cells expressing the feature in at least one
of the two groups, currently only used for poisson and negative binomial tests</p></dd>


<dt>min.cells.group</dt>
<dd><p>Minimum number of cells in one of the groups</p></dd>


<dt>norm.method</dt>
<dd><p>Normalization method for fold change calculation when slot is 'data'. Default is "LogNormalize".</p></dd>


<dt>p.adjust.method</dt>
<dd><p>A character value specifying the method to use for adjusting p-values. Default is "bonferroni".</p></dd>


<dt>slot</dt>
<dd><p>Slot to pull data from; note that if <code>test.use</code> is "negbinom", "poisson", or "DESeq2",
<code>slot</code> will be set to "counts"</p></dd>


<dt>assay</dt>
<dd><p>Assay to use in differential expression testing</p></dd>


<dt>BPPARAM</dt>
<dd><p>A BiocParallelParam object specifying the parallelization parameters for the differential test. Default is BiocParallel::bpparam().</p></dd>


<dt>seed</dt>
<dd><p>An integer value specifying the seed. Default is 11.</p></dd>


<dt>verbose</dt>
<dd><p>A logical value indicating whether to display progress messages during the differential test. Default is TRUE.</p></dd>


<dt>...</dt>
<dd><p>Additional arguments to pass to the <code><a href="https://satijalab.org/seurat/reference/FindMarkers.html" class="external-link">FindMarkers</a></code> function.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="RunEnrichment.html">RunEnrichment</a></code> <code><a href="RunGSEA.html">RunGSEA</a></code> <code><a href="GroupHeatmap.html">GroupHeatmap</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pancreas_sub"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunDEtest</span><span class="op">(</span><span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"SubCellType"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Data in the 'data' slot is raw counts. Perform NormalizeData(LogNormalize) on the data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-10-26 14:23:30.817226] Start DEtest</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Find all markers(wilcox) among 8 groups...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-10-26 14:23:35.73216] DEtest done</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Elapsed time:4.91 secs</span>
<span class="r-in"><span><span class="va">AllMarkers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pancreas_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_SubCellType</span><span class="op">$</span><span class="va">AllMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">AllMarkers</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Ductal   Ngn3 low EP  Ngn3 high EP Pre-endocrine          Beta         Alpha         Delta       Epsilon </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            75            29            69            39            61            50            37            44 </span>
<span class="r-in"><span><span class="va">ht1</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, features <span class="op">=</span> <span class="va">AllMarkers</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">AllMarkers</span><span class="op">$</span><span class="va">group1</span>, group.by <span class="op">=</span> <span class="st">"SubCellType"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'magick' package is suggested to install to give better rasterization.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Set `ht_opt$message = FALSE` to turn off this message.</span>
<span class="r-in"><span><span class="va">ht1</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-plt img"><img src="RunDEtest-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">TopMarkers</span> <span class="op">&lt;-</span> <span class="va">AllMarkers</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">avg_log2FC</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">group1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/top_n.html" class="external-link">top_n</a></span><span class="op">(</span><span class="fl">3</span>, <span class="va">avg_log2FC</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ht2</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, features <span class="op">=</span> <span class="va">TopMarkers</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">TopMarkers</span><span class="op">$</span><span class="va">group1</span>, group.by <span class="op">=</span> <span class="st">"SubCellType"</span>, show_row_names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'magick' package is suggested to install to give better rasterization.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Set `ht_opt$message = FALSE` to turn off this message.</span>
<span class="r-in"><span><span class="va">ht2</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-plt img"><img src="RunDEtest-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunDEtest</span><span class="op">(</span><span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"SubCellType"</span>, markers_type <span class="op">=</span> <span class="st">"paired"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-10-26 14:23:38.015016] Start DEtest</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Find paired markers(wilcox) among 8 groups...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-10-26 14:23:58.335474] DEtest done</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Elapsed time:20.32 secs</span>
<span class="r-in"><span><span class="va">PairedMarkers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pancreas_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_SubCellType</span><span class="op">$</span><span class="va">PairedMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">PairedMarkers</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Ductal   Ngn3 low EP  Ngn3 high EP Pre-endocrine          Beta         Alpha         Delta       Epsilon </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           893           787           680           571           645           575           426           575 </span>
<span class="r-in"><span><span class="va">ht3</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, features <span class="op">=</span> <span class="va">PairedMarkers</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">PairedMarkers</span><span class="op">$</span><span class="va">group1</span>, group.by <span class="op">=</span> <span class="st">"SubCellType"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'magick' package is suggested to install to give better rasterization.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Set `ht_opt$message = FALSE` to turn off this message.</span>
<span class="r-in"><span><span class="va">ht3</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-plt img"><img src="RunDEtest-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"panc8_sub"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">panc8_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="Integration_SCP.html">Integration_SCP</a></span><span class="op">(</span><span class="va">panc8_sub</span>, batch <span class="op">=</span> <span class="st">"tech"</span>, integration_method <span class="op">=</span> <span class="st">"Seurat"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:06.873903] Start Seurat_integrate</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:06.885801] Spliting srtMerge into srtList by column tech... ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:07.333775] Checking srtList... ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Data 1/5 of the srtList is raw_normalized_counts. Perform NormalizeData(LogNormalize) on the data ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Perform FindVariableFeatures on the data 1/5 of the srtList...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Data 2/5 of the srtList is raw_normalized_counts. Perform NormalizeData(LogNormalize) on the data ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Perform FindVariableFeatures on the data 2/5 of the srtList...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Data 3/5 of the srtList is raw_normalized_counts. Perform NormalizeData(LogNormalize) on the data ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Perform FindVariableFeatures on the data 3/5 of the srtList...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Data 4/5 of the srtList is raw_counts. Perform NormalizeData(LogNormalize) on the data ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Perform FindVariableFeatures on the data 4/5 of the srtList...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Data 5/5 of the srtList is raw_counts. Perform NormalizeData(LogNormalize) on the data ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Perform FindVariableFeatures on the data 5/5 of the srtList...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Use the separate HVF from srtList...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Number of available HVF: 2000</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:10.068593] Finished checking.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:10.389293] Perform FindIntegrationAnchors on the data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:31.32992] Perform integration(Seurat) on the data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:38.456233] Perform ScaleData on the data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:38.70372] Perform linear dimension reduction (pca) on the data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:39.808603] Perform FindClusters (louvain) on the data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:39.970694] Reorder clusters...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:40.1562] Perform nonlinear dimension reduction (umap) on the data...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Non-linear dimensionality reduction(umap) using Reduction(Seuratpca, dims:1-12) as input</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Non-linear dimensionality reduction(umap) using Reduction(Seuratpca, dims:1-12) as input</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:50.566344] Seurat_integrate done</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Elapsed time: 43.69 secs </span>
<span class="r-in"><span><span class="fu"><a href="CellDimPlot.html">CellDimPlot</a></span><span class="op">(</span><span class="va">panc8_sub</span>, group.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"celltype"</span>, <span class="st">"tech"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="RunDEtest-4.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">panc8_sub</span> <span class="op">&lt;-</span> <span class="fu">RunDEtest</span><span class="op">(</span><span class="va">panc8_sub</span>, group_by <span class="op">=</span> <span class="st">"celltype"</span>, grouping.var <span class="op">=</span> <span class="st">"tech"</span>, markers_type <span class="op">=</span> <span class="st">"conserved"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-10-26 14:24:51.334954] Start DEtest</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Find conserved markers(wilcox) among 13 groups...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>celseq has fewer than 3 cells. Skipping celseq</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>celseq has fewer than 3 cells. Skipping celseq</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>celseq2 has fewer than 3 cells. Skipping celseq2</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>fluidigmc1 has fewer than 3 cells. Skipping fluidigmc1</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>smartseq2 has fewer than 3 cells. Skipping smartseq2</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>indrop has fewer than 3 cells. Skipping indrop</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>No group was tested</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>celseq has fewer than 3 cells. Skipping celseq</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>celseq2 has fewer than 3 cells. Skipping celseq2</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>smartseq2 has fewer than 3 cells. Skipping smartseq2</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>celseq has fewer than 3 cells. Skipping celseq</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>fluidigmc1 has fewer than 3 cells. Skipping fluidigmc1</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>smartseq2 has fewer than 3 cells. Skipping smartseq2</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>celseq has fewer than 3 cells. Skipping celseq</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>celseq2 has fewer than 3 cells. Skipping celseq2</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>fluidigmc1 has fewer than 3 cells. Skipping fluidigmc1</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>smartseq2 has fewer than 3 cells. Skipping smartseq2</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Only a single group was tested</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>celseq has fewer than 3 cells. Skipping celseq</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>celseq2 has fewer than 3 cells. Skipping celseq2</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>smartseq2 has fewer than 3 cells. Skipping smartseq2</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>indrop has fewer than 3 cells. Skipping indrop</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Only a single group was tested</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-10-26 14:25:21.417648] DEtest done</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Elapsed time:30.08 secs</span>
<span class="r-in"><span><span class="va">ConservedMarkers1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">panc8_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_celltype</span><span class="op">$</span><span class="va">ConservedMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">ConservedMarkers1</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              alpha               beta             ductal             acinar activated_stellate              gamma </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 34                 47                145                 89                171                 16 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              delta            epsilon               mast        endothelial         macrophage quiescent_stellate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  8                  0                  0                128                152                105 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            schwann </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 16 </span>
<span class="r-in"><span><span class="va">ht4</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">panc8_sub</span>,</span></span>
<span class="r-in"><span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span></span>
<span class="r-in"><span>  features <span class="op">=</span> <span class="va">ConservedMarkers1</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">ConservedMarkers1</span><span class="op">$</span><span class="va">group1</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"tech"</span>, split.by <span class="op">=</span> <span class="st">"celltype"</span>, within_groups <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'magick' package is suggested to install to give better rasterization.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Set `ht_opt$message = FALSE` to turn off this message.</span>
<span class="r-in"><span><span class="va">ht4</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-plt img"><img src="RunDEtest-5.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">panc8_sub</span> <span class="op">&lt;-</span> <span class="fu">RunDEtest</span><span class="op">(</span><span class="va">panc8_sub</span>, group_by <span class="op">=</span> <span class="st">"tech"</span>, grouping.var <span class="op">=</span> <span class="st">"celltype"</span>, markers_type <span class="op">=</span> <span class="st">"conserved"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-10-26 14:25:28.742674] Start DEtest</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Find conserved markers(wilcox) among 5 groups...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>mast has fewer than 3 cells. Skipping mast</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>schwann has fewer than 3 cells. Skipping schwann</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>epsilon has fewer than 3 cells. Skipping epsilon</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>macrophage has fewer than 3 cells. Skipping macrophage</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>quiescent_stellate has fewer than 3 cells. Skipping quiescent_stellate</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>endothelial has fewer than 3 cells. Skipping endothelial</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>activated_stellate has fewer than 3 cells. Skipping activated_stellate</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>mast has fewer than 3 cells. Skipping mast</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>schwann has fewer than 3 cells. Skipping schwann</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>epsilon has fewer than 3 cells. Skipping epsilon</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>quiescent_stellate has fewer than 3 cells. Skipping quiescent_stellate</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>endothelial has fewer than 3 cells. Skipping endothelial</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>mast has fewer than 3 cells. Skipping mast</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>schwann has fewer than 3 cells. Skipping schwann</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>epsilon has fewer than 3 cells. Skipping epsilon</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>macrophage has fewer than 3 cells. Skipping macrophage</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>quiescent_stellate has fewer than 3 cells. Skipping quiescent_stellate</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>mast has fewer than 3 cells. Skipping mast</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>schwann has fewer than 3 cells. Skipping schwann</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>epsilon has fewer than 3 cells. Skipping epsilon</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>mast has fewer than 3 cells. Skipping mast</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>schwann has fewer than 3 cells. Skipping schwann</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>epsilon has fewer than 3 cells. Skipping epsilon</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>macrophage has fewer than 3 cells. Skipping macrophage</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>quiescent_stellate has fewer than 3 cells. Skipping quiescent_stellate</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>endothelial has fewer than 3 cells. Skipping endothelial</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-10-26 14:25:51.661211] DEtest done</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Elapsed time:22.92 secs</span>
<span class="r-in"><span><span class="va">ConservedMarkers2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">panc8_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_tech</span><span class="op">$</span><span class="va">ConservedMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">ConservedMarkers2</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     celseq    celseq2 fluidigmc1     indrop  smartseq2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         29         11         61         52         22 </span>
<span class="r-in"><span><span class="va">ht4</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">panc8_sub</span>,</span></span>
<span class="r-in"><span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span></span>
<span class="r-in"><span>  features <span class="op">=</span> <span class="va">ConservedMarkers2</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">ConservedMarkers2</span><span class="op">$</span><span class="va">group1</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"tech"</span>, split.by <span class="op">=</span> <span class="st">"celltype"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'magick' package is suggested to install to give better rasterization.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Set `ht_opt$message = FALSE` to turn off this message.</span>
<span class="r-in"><span><span class="va">ht4</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-plt img"><img src="RunDEtest-6.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">panc8_sub</span> <span class="op">&lt;-</span> <span class="fu">RunDEtest</span><span class="op">(</span><span class="va">panc8_sub</span>, group_by <span class="op">=</span> <span class="st">"celltype"</span>, grouping.var <span class="op">=</span> <span class="st">"tech"</span>, markers_type <span class="op">=</span> <span class="st">"disturbed"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-10-26 14:25:54.247059] Start DEtest</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Find disturbed markers(wilcox) among 13 groups...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-10-26 14:26:20.242668] DEtest done</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Elapsed time:26 secs</span>
<span class="r-in"><span><span class="va">DisturbedMarkers</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">panc8_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_celltype</span><span class="op">$</span><span class="va">DisturbedMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="va">avg_log2FC</span> <span class="op">&gt;</span> <span class="fl">1</span> <span class="op">&amp;</span> <span class="va">var1</span> <span class="op">==</span> <span class="st">"smartseq2"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">DisturbedMarkers</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              alpha               beta             ductal             acinar activated_stellate              gamma </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 37                117                 71                116                  5                  6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              delta            epsilon               mast        endothelial         macrophage quiescent_stellate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 16                  0                  0                  0                  0                  0 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            schwann </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  0 </span>
<span class="r-in"><span><span class="va">ht5</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">panc8_sub</span>,</span></span>
<span class="r-in"><span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span></span>
<span class="r-in"><span>  features <span class="op">=</span> <span class="va">DisturbedMarkers</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">DisturbedMarkers</span><span class="op">$</span><span class="va">group1</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"celltype"</span>, split.by <span class="op">=</span> <span class="st">"tech"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'magick' package is suggested to install to give better rasterization.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Set `ht_opt$message = FALSE` to turn off this message.</span>
<span class="r-in"><span><span class="va">ht5</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-plt img"><img src="RunDEtest-7.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">gene_specific</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">DisturbedMarkers</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">DisturbedMarkers_specific</span> <span class="op">&lt;-</span> <span class="va">DisturbedMarkers</span><span class="op">[</span><span class="va">DisturbedMarkers</span><span class="op">$</span><span class="va">gene</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">gene_specific</span>, <span class="op">]</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">DisturbedMarkers_specific</span><span class="op">$</span><span class="va">group1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              alpha               beta             ductal             acinar activated_stellate              gamma </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  3                 67                 27                 61                  4                  3 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              delta            epsilon               mast        endothelial         macrophage quiescent_stellate </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                 13                  0                  0                  0                  0                  0 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            schwann </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                  0 </span>
<span class="r-in"><span><span class="va">ht6</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">panc8_sub</span>,</span></span>
<span class="r-in"><span>  slot <span class="op">=</span> <span class="st">"data"</span>,</span></span>
<span class="r-in"><span>  features <span class="op">=</span> <span class="va">DisturbedMarkers_specific</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">DisturbedMarkers_specific</span><span class="op">$</span><span class="va">group1</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"celltype"</span>, split.by <span class="op">=</span> <span class="st">"tech"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'magick' package is suggested to install to give better rasterization.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Set `ht_opt$message = FALSE` to turn off this message.</span>
<span class="r-in"><span><span class="va">ht6</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-plt img"><img src="RunDEtest-8.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">ht7</span> <span class="op">&lt;-</span> <span class="fu"><a href="GroupHeatmap.html">GroupHeatmap</a></span><span class="op">(</span><span class="va">panc8_sub</span>,</span></span>
<span class="r-in"><span>  slot <span class="op">=</span> <span class="st">"data"</span>, aggregate_fun <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">expm1</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>  features <span class="op">=</span> <span class="va">DisturbedMarkers_specific</span><span class="op">$</span><span class="va">gene</span>, feature_split <span class="op">=</span> <span class="va">DisturbedMarkers_specific</span><span class="op">$</span><span class="va">group1</span>,</span></span>
<span class="r-in"><span>  group.by <span class="op">=</span> <span class="st">"celltype"</span>, grouping.var <span class="op">=</span> <span class="st">"tech"</span>, numerator <span class="op">=</span> <span class="st">"smartseq2"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>When 'grouping.var' is specified, 'exp_method' can only be 'log2fc'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'magick' package is suggested to install to give better rasterization.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Set `ht_opt$message = FALSE` to turn off this message.</span>
<span class="r-in"><span><span class="va">ht7</span><span class="op">$</span><span class="va">plot</span></span></span>
<span class="r-plt img"><img src="RunDEtest-9.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Hao Zhang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

