<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Perform the enrichment analysis (over-representation) on the genes"><title>Perform the enrichment analysis (over-representation) on the genes — RunEnrichment • SCP</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Perform the enrichment analysis (over-representation) on the genes — RunEnrichment"><meta property="og:description" content="Perform the enrichment analysis (over-representation) on the genes"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">SCP</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.5.6</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/zhanghao-njmu/SCP">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Perform the enrichment analysis (over-representation) on the genes</h1>
      <small class="dont-index">Source: <a href="https://github.com/zhanghao-njmu/SCP/blob/HEAD/R/SCP-analysis.R" class="external-link"><code>R/SCP-analysis.R</code></a></small>
      <div class="d-none name"><code>RunEnrichment.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Perform the enrichment analysis (over-representation) on the genes</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">RunEnrichment</span><span class="op">(</span></span>
<span>  srt <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  group_by <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  test.use <span class="op">=</span> <span class="st">"wilcox"</span>,</span>
<span>  DE_threshold <span class="op">=</span> <span class="st">"avg_log2FC &gt; 0 &amp; p_val_adj &lt; 0.05"</span>,</span>
<span>  geneID <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  geneID_groups <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  geneID_exclude <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  IDtype <span class="op">=</span> <span class="st">"symbol"</span>,</span>
<span>  result_IDtype <span class="op">=</span> <span class="st">"symbol"</span>,</span>
<span>  species <span class="op">=</span> <span class="st">"Homo_sapiens"</span>,</span>
<span>  db <span class="op">=</span> <span class="st">"GO_BP"</span>,</span>
<span>  db_update <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  db_version <span class="op">=</span> <span class="st">"latest"</span>,</span>
<span>  db_combine <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  convert_species <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  Ensembl_version <span class="op">=</span> <span class="fl">103</span>,</span>
<span>  mirror <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  TERM2GENE <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  TERM2NAME <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  minGSSize <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  maxGSSize <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  unlimited_db <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Chromosome"</span>, <span class="st">"GeneType"</span>, <span class="st">"TF"</span>, <span class="st">"Enzyme"</span>, <span class="st">"CSPA"</span><span class="op">)</span>,</span>
<span>  GO_simplify <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  GO_simplify_cutoff <span class="op">=</span> <span class="st">"p.adjust &lt; 0.05"</span>,</span>
<span>  simplify_method <span class="op">=</span> <span class="st">"Wang"</span>,</span>
<span>  simplify_similarityCutoff <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">bpparam</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">11</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>srt</dt>
<dd><p>A Seurat object containing the results of differential expression analysis (RunDEtest).
If specified, the genes and groups will be extracted from the Seurat object automatically.
If not specified, the <code>geneID</code> and <code>geneID_groups</code> arguments must be provided.</p></dd>


<dt>group_by</dt>
<dd><p>A character vector specifying the grouping variable in the Seurat object. This argument is only used if <code>srt</code> is specified.</p></dd>


<dt>test.use</dt>
<dd><p>A character vector specifying the test to be used in differential expression analysis. This argument is only used if <code>srt</code> is specified.</p></dd>


<dt>DE_threshold</dt>
<dd><p>A character vector specifying the filter condition for differential expression analysis. This argument is only used if <code>srt</code> is specified.</p></dd>


<dt>geneID</dt>
<dd><p>A character vector specifying the gene IDs.</p></dd>


<dt>geneID_groups</dt>
<dd><p>A factor vector specifying the group labels for each gene.</p></dd>


<dt>geneID_exclude</dt>
<dd><p>A character vector specifying the gene IDs to be excluded from the analysis.</p></dd>


<dt>IDtype</dt>
<dd><p>A character vector specifying the type of gene IDs in the <code>srt</code> object or <code>geneID</code> argument. This argument is used to convert the gene IDs to a different type if <code>IDtype</code> is different from <code>result_IDtype</code>.</p></dd>


<dt>result_IDtype</dt>
<dd><p>A character vector specifying the desired type of gene ID to be used in the output. This argument is used to convert the gene IDs from <code>IDtype</code> to <code>result_IDtype</code>.</p></dd>


<dt>species</dt>
<dd><p>A character vector specifying the species for which the analysis is performed.</p></dd>


<dt>db</dt>
<dd><p>A character vector specifying the name of the database to be used for enrichment analysis.</p></dd>


<dt>db_update</dt>
<dd><p>A logical value indicating whether the gene annotation databases should be forcefully updated. If set to FALSE, the function will attempt to load the cached databases instead. Default is FALSE.</p></dd>


<dt>db_version</dt>
<dd><p>A character vector specifying the version of the database to be used. This argument is ignored if <code>db_update</code> is <code>TRUE</code>. Default is "latest".</p></dd>


<dt>db_combine</dt>
<dd><p>A logical value indicating whether to combine multiple databases into one. If TRUE, all database specified by <code>db</code> will be combined as one named "Combined".</p></dd>


<dt>convert_species</dt>
<dd><p>A logical value indicating whether to use a species-converted database when the annotation is missing for the specified species. The default value is TRUE.</p></dd>


<dt>Ensembl_version</dt>
<dd><p>Ensembl database version. If NULL, use the current release version.</p></dd>


<dt>mirror</dt>
<dd><p>Specify an Ensembl mirror to connect to. The valid options here are 'www', 'uswest', 'useast', 'asia'.</p></dd>


<dt>TERM2GENE</dt>
<dd><p>A data frame specifying the gene-term mapping for a custom database. The first column should contain the term IDs, and the second column should contain the gene IDs.</p></dd>


<dt>TERM2NAME</dt>
<dd><p>A data frame specifying the term-name mapping for a custom database. The first column should contain the term IDs, and the second column should contain the corresponding term names.</p></dd>


<dt>minGSSize</dt>
<dd><p>A numeric value specifying the minimum size of a gene set to be considered in the enrichment analysis.</p></dd>


<dt>maxGSSize</dt>
<dd><p>A numeric value specifying the maximum size of a gene set to be considered in the enrichment analysis.</p></dd>


<dt>unlimited_db</dt>
<dd><p>A character vector specifying the names of databases that do not have size restrictions.</p></dd>


<dt>GO_simplify</dt>
<dd><p>A logical value indicating whether to simplify the GO terms. If <code>TRUE</code>, additional results with simplified GO terms will be returned.</p></dd>


<dt>GO_simplify_cutoff</dt>
<dd><p>A character vector specifying the filter condition for simplification of GO terms. This argument is only used if <code>GO_simplify</code> is <code>TRUE</code>.</p></dd>


<dt>simplify_method</dt>
<dd><p>A character vector specifying the method to be used for simplification of GO terms. This argument is only used if <code>GO_simplify</code> is <code>TRUE</code>.</p></dd>


<dt>simplify_similarityCutoff</dt>
<dd><p>A numeric value specifying the similarity cutoff for simplification of GO terms. This argument is only used if <code>GO_simplify</code> is <code>TRUE</code>.</p></dd>


<dt>BPPARAM</dt>
<dd><p>A BiocParallelParam object specifying the parallel back-end to be used for parallel computation. Defaults to BiocParallel::bpparam().</p></dd>


<dt>seed</dt>
<dd><p>The random seed for reproducibility. Defaults to 11.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>If input is a Seurat object, returns the modified Seurat object with the enrichment result stored in the tools slot.</p>


<p>If input is a geneID vector with or without geneID_groups, return the enrichment result directly.</p>


<p>Enrichment result is a list with the following component:</p><ul><li><p><code>enrichment</code>: A data.frame containing all enrichment results.</p></li>
<li><p><code>results</code>: A list of <code>enrichResult</code> objects from the DOSE package.</p></li>
<li><p><code>geneMap</code>: A data.frame containing the ID mapping table for input gene IDs.</p></li>
<li><p><code>input</code>: A data.frame containing the input gene IDs and gene ID groups.</p></li>
<li><p><code>DE_threshold</code>: A specific threshold for differential expression analysis (only returned if input is a Seurat object).</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="PrepareDB.html">PrepareDB</a></code> <code><a href="ListDB.html">ListDB</a></code> <code><a href="EnrichmentPlot.html">EnrichmentPlot</a></code> <code><a href="RunGSEA.html">RunGSEA</a></code> <code><a href="GSEAPlot.html">GSEAPlot</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"pancreas_sub"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu"><a href="RunDEtest.html">RunDEtest</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Data in the 'data' slot is raw counts. Perform NormalizeData(LogNormalize) on the data.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-11-20 16:43:57.641219] Start DEtest</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Find all markers(wilcox) among 5 groups...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |                                                                                                              |   0%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |============================================                                                                  |  40%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                                                                      |==============================================================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-11-20 16:44:08.200234] DEtest done</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Elapsed time:10.56 secs</span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunEnrichment</span><span class="op">(</span></span></span>
<span class="r-in"><span>  srt <span class="op">=</span> <span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, DE_threshold <span class="op">=</span> <span class="st">"p_val_adj &lt; 0.05"</span>,</span></span>
<span class="r-in"><span>  db <span class="op">=</span> <span class="st">"GO_BP"</span>, species <span class="op">=</span> <span class="st">"Mus_musculus"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-11-20 16:44:08.204245] Start Enrichment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Species: Mus_musculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading cached db: GO_BP version:3.17.0 nterm:16027 created:2023-11-20 16:20:38.408212</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Convert ID types for the database: GO_BP</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connect to the Ensembl archives...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the 103 version of biomart...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connecting to the biomart...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Searching the dataset mmusculus ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connecting to the dataset mmusculus_gene_ensembl ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting the geneIDs...</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in collect(., Inf):</span> Failed to collect lazy table.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Caused by error in `db_collect()`:</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Arguments in `...` must be used.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Problematic argument:</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> ..1 = Inf</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span>
<span class="r-in"><span><span class="fu"><a href="EnrichmentPlot.html">EnrichmentPlot</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, plot_type <span class="op">=</span> <span class="st">"comparison"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in EnrichmentPlot(pancreas_sub, db = "GO_BP", group_by = "CellType",     plot_type = "comparison"):</span> No enrichment result found. You may perform RunEnrichment first.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunEnrichment</span><span class="op">(</span></span></span>
<span class="r-in"><span>  srt <span class="op">=</span> <span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, DE_threshold <span class="op">=</span> <span class="st">"p_val_adj &lt; 0.05"</span>,</span></span>
<span class="r-in"><span>  db <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MSigDB"</span>, <span class="st">"MSigDB_MH"</span><span class="op">)</span>, species <span class="op">=</span> <span class="st">"Mus_musculus"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-11-20 16:44:12.325598] Start Enrichment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Species: Mus_musculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing database: MSigDB</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Convert ID types for the database: MSigDB</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connect to the Ensembl archives...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the 103 version of biomart...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connecting to the biomart...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Searching the dataset mmusculus ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connecting to the dataset mmusculus_gene_ensembl ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting the geneIDs...</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in collect(., Inf):</span> Failed to collect lazy table.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Caused by error in `db_collect()`:</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Arguments in `...` must be used.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Problematic argument:</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> ..1 = Inf</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span>
<span class="r-in"><span><span class="fu"><a href="EnrichmentPlot.html">EnrichmentPlot</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, db <span class="op">=</span> <span class="st">"MSigDB"</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, plot_type <span class="op">=</span> <span class="st">"comparison"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in EnrichmentPlot(pancreas_sub, db = "MSigDB", group_by = "CellType",     plot_type = "comparison"):</span> No enrichment result found. You may perform RunEnrichment first.</span>
<span class="r-in"><span><span class="fu"><a href="EnrichmentPlot.html">EnrichmentPlot</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, db <span class="op">=</span> <span class="st">"MSigDB_MH"</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, plot_type <span class="op">=</span> <span class="st">"comparison"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in EnrichmentPlot(pancreas_sub, db = "MSigDB_MH", group_by = "CellType",     plot_type = "comparison"):</span> No enrichment result found. You may perform RunEnrichment first.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Remove redundant GO terms</span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunEnrichment</span><span class="op">(</span>srt <span class="op">=</span> <span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span>, GO_simplify <span class="op">=</span> <span class="cn">TRUE</span>, species <span class="op">=</span> <span class="st">"Mus_musculus"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-11-20 16:44:27.916924] Start Enrichment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Species: Mus_musculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading cached db: GO_BP version:3.17.0 nterm:16027 created:2023-11-20 16:20:38.408212</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Convert ID types for the database: GO_BP</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connect to the Ensembl archives...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the 103 version of biomart...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connecting to the biomart...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Searching the dataset mmusculus ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connecting to the dataset mmusculus_gene_ensembl ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting the geneIDs...</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in collect(., Inf):</span> Failed to collect lazy table.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Caused by error in `db_collect()`:</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Arguments in `...` must be used.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Problematic argument:</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> ..1 = Inf</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span>
<span class="r-in"><span><span class="fu"><a href="EnrichmentPlot.html">EnrichmentPlot</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, db <span class="op">=</span> <span class="st">"GO_BP_sim"</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, plot_type <span class="op">=</span> <span class="st">"comparison"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in EnrichmentPlot(pancreas_sub, db = "GO_BP_sim", group_by = "CellType",     plot_type = "comparison"):</span> No enrichment result found. You may perform RunEnrichment first.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Use a combined database</span></span></span>
<span class="r-in"><span><span class="va">pancreas_sub</span> <span class="op">&lt;-</span> <span class="fu">RunEnrichment</span><span class="op">(</span></span></span>
<span class="r-in"><span>  srt <span class="op">=</span> <span class="va">pancreas_sub</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>,</span></span>
<span class="r-in"><span>  db <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"KEGG"</span>, <span class="st">"WikiPathway"</span>, <span class="st">"Reactome"</span>, <span class="st">"PFAM"</span>, <span class="st">"MP"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  db_combine <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>  species <span class="op">=</span> <span class="st">"Mus_musculus"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-11-20 16:44:31.848021] Start Enrichment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Species: Mus_musculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading cached db: MP version:2023-11-20 08:08 nterm:10311 created:2023-11-20 16:20:58.084545</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing database: KEGG</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing database: WikiPathway</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing database: Reactome</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Preparing database: PFAM</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Convert ID types for the database: KEGG</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connect to the Ensembl archives...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the 103 version of biomart...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connecting to the biomart...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Searching the dataset mmusculus ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connecting to the dataset mmusculus_gene_ensembl ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting the geneIDs...</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in collect(., Inf):</span> Failed to collect lazy table.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Caused by error in `db_collect()`:</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Arguments in `...` must be used.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Problematic argument:</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> ..1 = Inf</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span>
<span class="r-in"><span><span class="fu"><a href="EnrichmentPlot.html">EnrichmentPlot</a></span><span class="op">(</span><span class="va">pancreas_sub</span>, db <span class="op">=</span> <span class="st">"Combined"</span>, group_by <span class="op">=</span> <span class="st">"CellType"</span>, plot_type <span class="op">=</span> <span class="st">"comparison"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in EnrichmentPlot(pancreas_sub, db = "Combined", group_by = "CellType",     plot_type = "comparison"):</span> No enrichment result found. You may perform RunEnrichment first.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Or use "geneID" and "geneID_groups" as input to run enrichment</span></span></span>
<span class="r-in"><span><span class="va">de_df</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pancreas_sub</span><span class="op">@</span><span class="va">tools</span><span class="op">$</span><span class="va">DEtest_CellType</span><span class="op">$</span><span class="va">AllMarkers_wilcox</span>, <span class="va">p_val_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">enrich_out</span> <span class="op">&lt;-</span> <span class="fu">RunEnrichment</span><span class="op">(</span>geneID <span class="op">=</span> <span class="va">de_df</span><span class="op">[[</span><span class="st">"gene"</span><span class="op">]</span><span class="op">]</span>, geneID_groups <span class="op">=</span> <span class="va">de_df</span><span class="op">[[</span><span class="st">"group1"</span><span class="op">]</span><span class="op">]</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span>, species <span class="op">=</span> <span class="st">"Mus_musculus"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> [2023-11-20 16:45:02.209212] Start Enrichment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Workers: 2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Species: Mus_musculus</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading cached db: GO_BP version:3.17.0 nterm:16027 created:2023-11-20 16:20:38.408212</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Convert ID types for the database: GO_BP</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connect to the Ensembl archives...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the 103 version of biomart...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connecting to the biomart...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Searching the dataset mmusculus ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Connecting to the dataset mmusculus_gene_ensembl ...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Converting the geneIDs...</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in collect(., Inf):</span> Failed to collect lazy table.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="font-weight: bold;">Caused by error in `db_collect()`:</span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Arguments in `...` must be used.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Problematic argument:</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> ..1 = Inf</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Did you misspell an argument name?</span>
<span class="r-in"><span><span class="fu"><a href="EnrichmentPlot.html">EnrichmentPlot</a></span><span class="op">(</span>res <span class="op">=</span> <span class="va">enrich_out</span>, db <span class="op">=</span> <span class="st">"GO_BP"</span>, plot_type <span class="op">=</span> <span class="st">"comparison"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in eval(expr, envir, enclos):</span> object 'enrich_out' not found</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Hao Zhang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

